### PR TITLE
Added Static Metric Generation Wokflow

### PR DESCRIPTION
**Pull request by akashest41**
### **Adding `MetricHandler` class for handling and generating metrics**

A new `MetricHandler` class is now available to generate a list of static metrics on-the-fly for optimizing the agent for the provided task description. This class is now instantiated inside the agent optimization pipeline. See `/metric_generator/` for more. Besides requiring a task description, the user can also edit the `num_metrics` parameter during configuration to request a specific number of generated metrics. **Shout-out to Elad for helping with these!**

### **Added meta prompts for handling the metric generation**

A new meta prompt is now available under `/prompts/meta_prompts_generation/` for creating helpful metrics. 

### **Added JSON Schema for generated metrics**

A new JSON schema has been added to provide a format for metric generation. The generated metrics currently have the following general format:
```
{
"metrics_list": [
        {
            "metric_name": "...",
            "metric_desc": "...",
            "metric_prompt": "...",
            "metric_scale": [
                {
                    "deficient_desc": "...",
                    "adequate_desc": "...",
                    "competent_desc": "...",
                    "proficient_desc": "...",
                    "exemplary_desc": "..."
                }
            ]
        }, ...
    ]
}
```

**Shout-out to Elad and Michael for feedback and helpful discussions here!**

### CODE REVIEW

File: metric_generator/.DS_Store
Comment (Eladlev) [inline metric_generator/.DS_Store:1]: This file should be removed from the PR

File: requirements.txt [hunk 1]
```diff
@@ -14,3 +14,4 @@ faiss-cpu==1.7.4
 sentence-transformers==2.2.2
 langchain-google-genai==0.0.9
 pillow==10.2.0
+langchain_openai==0.1.20
```
Comment (Eladlev) [inline requirements.txt:17]: need to add it also to the conda installation environment_dev.yml

**Review by Eladlev — CHANGES_REQUESTED** [review]
Looking great, only left 2 minor comments

File: metric_generator/.DS_Store
Comment (akashest41) [reply metric_generator/.DS_Store:1]: should be done now!

File: requirements.txt (hunk 1 above)
Comment (akashest41) [reply requirements.txt:17]: included this now

**Pull request merged**